Loop: monitor cmd crashes with nil pointer error

### Background
Attempting to loop out funds from a channel, when entering the command `loop monitor`, loopd crashes with a nil pointer exception. (The loop out was still successful).

### Your environment
LND v0.5.2
Linux version 4.18.0-10-generic (buildd@lgw01-amd64-060) (gcc version 8.2.0 (Ubuntu 8.2.0-7ubuntu1))

### Steps to reproduce
1. Run loopd : `loopd --network=testnet`
2. Loop out: `loop out --channel=<> --amt=<>`
3. Run monitor: `loop monitor`

### Expected Behaviour
Be able to run monitor and view the progress of activity swaps

### Actual Behaviour
Nil pointer exception

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xa8b0bc]

goroutine 57 [running]:
main.(*swapClientServer).marshallSwap(0xc00003b1d0, 0xc00030d3b0, 0xc0001ba600, 0xb05fc0, 0xc0001ea1c0)
        /loop/cmd/loopd/swapclient_server.go:126 +0x6bc
main.(*swapClientServer).Monitor.func1(0x208245de, 0xed43b593c, 0x13ee2e0, 0x2db4b5376f277fe7, 0x59e029497dedf7cc, 0xc403cd081975b1e0, 0xfab4f59ad8094b17, 0x102, 0xbf2cc6d465218e97, 0x2417b79d917bbdd9, ...)
        /loop/cmd/loopd/swapclient_server.go:145 +0x4a
main.(*swapClientServer).Monitor(0xc00003b1d0, 0xc0001ea100, 0xe69380, 0xc0002040d0, 0x0, 0x0)
        /loop/cmd/loopd/swapclient_server.go:208 +0x8c6
github.com/lightninglabs/loop/looprpc._SwapClient_Monitor_Handler(0xbebe80, 0xc00003b1d0, 0xe68ba0, 0xc0000300c0, 0x140d708, 0xc0001ba4e0)
        /loop/looprpc/client.pb.go:975 +0x109
google.golang.org/grpc.(*Server).processStreamingRPC(0xc000087b00, 0xe6aee0, 0xc0002ee900, 0xc00026c100, 0xc00028e540, 0x13d68c0, 0x0, 0x0, 0x0)
        /go/pkg/mod/google.golang.org/grpc@v1.19.0/server.go:1175 +0xac7
google.golang.org/grpc.(*Server).handleStream(0xc000087b00, 0xe6aee0, 0xc0002ee900, 0xc00026c100, 0x0)
        /go/pkg/mod/google.golang.org/grpc@v1.19.0/server.go:1254 +0xcf1
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000018020, 0xc000087b00, 0xe6aee0, 0xc0002ee900, 0xc00026c100)
        /go/pkg/mod/google.golang.org/grpc@v1.19.0/server.go:690 +0x9f
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /go/pkg/mod/google.golang.org/grpc@v1.19.0/server.go:688 +0xa1
```

### Additional Information

After debug printing, it seems the HtlcAddress is nil

```
2019-04-07 03:46:43.002 [INF] LOOPD: IS IT STATE: SUCCESS
2019-04-07 03:46:43.002 [INF] LOOPD: IS IT Initiation Time: 1554602920576774837
2019-04-07 03:46:43.002 [INF] LOOPD: IS IT LastUpdateTime: 1554604604545408478
2019-04-07 03:46:43.002 [INF] LOOPD: IS IT HtlcAddress: <nil>
2019-04-07 03:46:43.002 [INF] LOOPD: IS IT type: LOOP_OUT
```

In `cmd/loopd/swapclient_server.go` in func `marshallSwap`, the return `*looprpc.SwapStatus` calls `loopSwap.HtlcAddress.EncodeAddress()` on nil.

--- a/client.go
+++ b/client.go
@@ -113,14 +113,61 @@ func NewClient(dbDir string, serverAddress string, insecure bool,
 	return client, cleanup, nil
 }
 
-// FetchLoopOutSwaps returns a list of all swaps currently in the database.
-func (s *Client) FetchLoopOutSwaps() ([]*loopdb.LoopOut, error) {
-	return s.Store.FetchLoopOutSwaps()
-}
+// FetchSwaps returns all loop in and out swaps currently in the database.
+func (s *Client) FetchSwaps() ([]*SwapInfo, error) {
+	loopOutSwaps, err := s.Store.FetchLoopOutSwaps()
+	if err != nil {
+		return nil, err
+	}
 
-// FetchLoopInSwaps returns a list of all swaps currently in the database.
-func (s *Client) FetchLoopInSwaps() ([]*loopdb.LoopIn, error) {
-	return s.Store.FetchLoopInSwaps()
+	loopInSwaps, err := s.Store.FetchLoopInSwaps()
+	if err != nil {
+		return nil, err
+	}
+
+	swaps := make([]*SwapInfo, 0, len(loopInSwaps)+len(loopOutSwaps))
+
+	for _, swp := range loopOutSwaps {
+		htlc, err := swap.NewHtlc(
+			swp.Contract.CltvExpiry, swp.Contract.SenderKey,
+			swp.Contract.ReceiverKey, swp.Hash, swap.HtlcP2WSH,
+			s.lndServices.ChainParams,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		swaps = append(swaps, &SwapInfo{
+			SwapType:     TypeOut,
+			SwapContract: swp.Contract.SwapContract,
+			State:        swp.State(),
+			SwapHash:     swp.Hash,
+			LastUpdate:   swp.LastUpdateTime(),
+			HtlcAddress:  htlc.Address,
+		})
+	}
+
+	for _, swp := range loopInSwaps {
+		htlc, err := swap.NewHtlc(
+			swp.Contract.CltvExpiry, swp.Contract.SenderKey,
+			swp.Contract.ReceiverKey, swp.Hash, swap.HtlcNP2WSH,
+			s.lndServices.ChainParams,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		swaps = append(swaps, &SwapInfo{
+			SwapType:     TypeIn,
+			SwapContract: swp.Contract.SwapContract,
+			State:        swp.State(),
+			SwapHash:     swp.Hash,
+			LastUpdate:   swp.LastUpdateTime(),
+			HtlcAddress:  htlc.Address,
+		})
+	}
+
+	return swaps, nil
 }
 
 // Run is a blocking call that executes all swaps. Any pending swaps are

--- a/cmd/loopd/daemon.go
+++ b/cmd/loopd/daemon.go
@@ -32,6 +32,7 @@ func daemon(config *config) error {
 		config.SwapServer = testnetServer
 	}
 
+	// Create an instance of the loop client library.
 	swapClient, cleanup, err := getClient(
 		config.Network, config.SwapServer, config.Insecure, &lnd.LndServices,
 	)
@@ -40,35 +41,14 @@ func daemon(config *config) error {
 	}
 	defer cleanup()
 
-	// Before starting the client, build an in-memory view of all swaps.
-	// This view is used to update newly connected clients with the most
-	// recent swaps.
-	loopOutSwaps, err := swapClient.FetchLoopOutSwaps()
+	// Retrieve all currently existing swaps from the database.
+	swapsList, err := swapClient.FetchSwaps()
 	if err != nil {
 		return err
-	}
-	for _, swap := range loopOutSwaps {
-		swaps[swap.Hash] = loop.SwapInfo{
-			SwapType:     loop.TypeOut,
-			SwapContract: swap.Contract.SwapContract,
-			State:        swap.State(),
-			SwapHash:     swap.Hash,
-			LastUpdate:   swap.LastUpdateTime(),
-		}
 	}
 
-	loopInSwaps, err := swapClient.FetchLoopInSwaps()
-	if err != nil {
-		return err
-	}
-	for _, swap := range loopInSwaps {
-		swaps[swap.Hash] = loop.SwapInfo{
-			SwapType:     loop.TypeIn,
-			SwapContract: swap.Contract.SwapContract,
-			State:        swap.State(),
-			SwapHash:     swap.Hash,
-			LastUpdate:   swap.LastUpdateTime(),
-		}
+	for _, s := range swapsList {
+		swaps[s.SwapHash] = *s
 	}
 
 	// Instantiate the loopd gRPC server.

--- a/cmd/loopd/view.go
+++ b/cmd/loopd/view.go
@@ -42,7 +42,7 @@ func view(config *config) error {
 }
 
 func viewOut(swapClient *loop.Client, chainParams *chaincfg.Params) error {
-	swaps, err := swapClient.FetchLoopOutSwaps()
+	swaps, err := swapClient.Store.FetchLoopOutSwaps()
 	if err != nil {
 		return err
 	}
@@ -88,7 +88,7 @@ func viewOut(swapClient *loop.Client, chainParams *chaincfg.Params) error {
 }
 
 func viewIn(swapClient *loop.Client, chainParams *chaincfg.Params) error {
-	swaps, err := swapClient.FetchLoopInSwaps()
+	swaps, err := swapClient.Store.FetchLoopInSwaps()
 	if err != nil {
 		return err
 	}

